File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/order/VirologyWebsiteConfig.java
package uk.nhs.nhsx.virology.order;

public class VirologyWebsiteConfig {
    
    public final String orderWebsite;
    public final String registerWebsite;

    public VirologyWebsiteConfig(String orderWebsite, String registerWebsite) {
        this.orderWebsite = orderWebsite;
        this.registerWebsite = registerWebsite;
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/order/VirologyWebsiteConfig.kt
package uk.nhs.nhsx.virology.order

import uk.nhs.nhsx.core.Environment
import uk.nhs.nhsx.core.Environment.EnvironmentKey

data class VirologyWebsiteConfig(val orderWebsite: String,
                                 val registerWebsite: String) {

    companion object {
        private val ORDER_WEBSITE = EnvironmentKey.string("order_website")
        private val REGISTER_WEBSITE = EnvironmentKey.string("register_website")

        fun fromEnvironment(environment: Environment): VirologyWebsiteConfig = VirologyWebsiteConfig(
            environment.access.required(ORDER_WEBSITE),
            environment.access.required(REGISTER_WEBSITE)
        )
    }
}
